orttraining/orttraining/python/training/optim/lr_scheduler.py

```diff
@@ -0,0 +1,290 @@
+import math
+
+from . import config
+
+
+class _LRScheduler(object):
+    r"""Base class for implementing custom learning rate schedulers
+
+    Schedulers can be either stateful or stateless.
+    Stateless implementations can only rely on information available at
+    :py:class:`.TrainStepInfo` and :py:attr:`.optimizer_config`.
+    Stateful, on the other hand, can store more parameters by overriding
+    the constructor.
```

Comment:
Could we claim that for a given initialized stateless scheduler, for a particular instance of `TrainStepInfo` object, the `get_lr` method will always output the same result. 

stateful, on the other hand, might return different results each time even called with the same `TrainStepInfo` object.

Reply:
Generally speaking, yes, as a consequence of stateless vs stateful definition. But It is possible that even stateless implementation can use random generatorts initialized by "current timestamp" and other stuff that are not stored anywhere, but can be used as an input for the scheduling algorithm

Removing all this randomness aforementioned, stateless implementation can output the same result given a static `TrainStepInfo` *and* `_OptimizerConfig`